src/inference/src/dev/core_impl.cpp

```diff
@@ -1077,6 +1077,16 @@ std::shared_ptr<const ov::Model> ov::CoreImpl::apply_auto_batching(const std::sh
         auto pos = deviceName.find_first_of(":");
         if (pos == std::string::npos)
             return model;  // BATCH device is already configured via the config
+
+        if (deviceName.find("(") == std::string::npos) {
+            auto supported_properties =
+                ICore::get_property(deviceName.substr(pos + 1, deviceName.find("(")), ov::supported_properties, {});
+            if (std::find(supported_properties.begin(), supported_properties.end(), ov::optimal_batch_size) ==
+                supported_properties.end())
+                return model;
+        } else {
+            deviceNameWithBatchSize = deviceName.substr(pos + 1);
+        }
         deviceNameWithBatchSize = deviceName.substr(pos + 1);
         deviceNameWithoutBatch = ov::DeviceIDParser::get_batch_device(deviceNameWithBatchSize);
```

Comment:
@ilya-lavrenov The "optimal batch size" inside () will be used in priority if it is provided, if is not provided, will be estimated here (including check optimal_batch_size): https://github.com/openvinotoolkit/openvino/blob/b43b9f9d5098a3702ce2a84b4a3eabfa39010d08/src/plugins/auto_batch/src/plugin.cpp#L240 . In the case of "optimal batch size" has been provided in parenthesis, optimal_batch_size property support is not required.

Reply:
> looks like `deviceNameWithoutBatch` is actual HW device we are trying to use. Can we check it for `optimal_batch_size` support instead of `deviceName.substr(pos + 1, deviceName.find("("))` ?

Looks like deviceNameWithoutBatch value is from user?  We have checked optimal_batch_size when batch size is not given.

Reply:
ok, can we avoid double parsing of HW device name? one is parsed in newly added code, second time we parse it in `ov::DeviceIDParser::get_batch_device`